### DISCUSSION TITLE
Module '"@odata2ts/http-client-api"' has no exported member 'ODataClientError'

### DISCUSSION THREAD

**Discussion by szarcensky** [odata2ts/odata2ts#227] [Q&A] [answered]
When compiling the generated code using `tsc` I'm getting this error:

```
node_modules/@odata2ts/http-client-fetch/lib/FetchClientError.d.ts:1:10 - 
error TS2305: Module '"@odata2ts/http-client-api"' has no exported member 'ODataClientError'.

import { ODataClientError } from "@odata2ts/http-client-api";
```

This is using the `test-ts-setup` repo, with any of the build scripts there. I've not changed any of the tsconfig files. Let me know if I need to post more specific info.

**Comment by texttechne** [comment]
Hi @szarcensky,

I've just checked out the repo, installed everything and executed `npm run build`. It fails with the UMD build part for a different reason.
Hence, I can't reproduce your issue.

But should have something to do with your dependencies. Have you installed different versions? Have you tried to nuke your node_modules folder and did a clean install?

**Reply by szarcensky** [reply]
Thanks for the reply, @texttechne ,

I've cloned a clean install of the repo. You don't see the issue with the repo because nothing in `/src` is attempting to import `FetchClient`. The tests aren't compiled. If you add:
`import { FetchClient } from '@odata2ts/http-client-fetch' `
to `src/index.ts` you will see the error. I can also see the error by opening `node_modules/@odata2ts/http-client-fetch/lib/FetchClientError.d.ts`

If I open `node_modules/@odata2ts/http-client-api/lib/` I see the following exported:
```
ODataHttpClientConfig<ClientType extends ODataHttpClient>
ODataHttpClient<RequestConfig = any>
HttpResponseModel<T>
ODataResponse<T> = Promise<HttpResponseModel<T>>
```
I don't see `ODataClientError` being exported, which is the error being thrown.

**Reply by szarcensky** [reply]
The UMD error is resolved by updating webpack to the current version, btw. You don't need the UMD version to work to see the error above, however.

**Comment by szarcensky** [comment, marked as answer]
So for some reason, npm was installing `v0.1.0` of `@odata2ts/http-client-api` even though the dependency says `^0.5.1`, and `npm ci` wasn't fixing it. I updated to the latest version of `@odata2ts` and that fixed it. From `0.29.1` to `0.30.0`.
